### PR TITLE
fix: calling programs with spaces on Windows

### PR DESCRIPTION
**Pull request by Kha**
This entire API has been carefully optimized for maximum pain output

### CODE REVIEW

File: src/runtime/process.cpp [hunk 1]
```diff
@@ -236,7 +236,9 @@ static obj_res spawn(string_ref const & proc_name, array_ref<string_ref> const &
 
     // Create the child process.
     bool bSuccess = CreateProcess(
-        program.c_str(),
+        // Passing `program` here should be more robust, but would require adding a `.exe` extension
+        // where necessary
+        NULL, //program.c_str(),
```
Comment (tydeu) [inline src/runtime/process.cpp:241]: According to the [docs](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa), this argument also does look in `PATH` (hence even `git.exe` would not work).

Comment (Kha) [reply src/runtime/process.cpp:241]: I assume you forgot a negation there

**Comment by Kha** [comment]
Well, *something* broke, but it seems more likely to be a Unicode problem with Lake (I think we're still not setting the code page to UTF-8) or something on the way to it.

File: tests/pkg/spaces/test.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+rm -rf .lake/build
+lake exe "«path with spaces»"
```
Comment (tydeu) [inline tests/pkg/spaces/test.sh:4]:
```suggestion
lake exe "path with spaces"
```

I think the French quotes here are causing the problem. I skip the part of Lake's `init` test that uses French quotes because I have found they are not properly passed to Lake from the CLI.

https://github.com/leanprover/lean4/blob/073b2cfc83dbb385548b5e5851e69d0713a578dc/src/lake/tests/init/test.sh#L124-L128

Comment (tydeu) [reply tests/pkg/spaces/test.sh:4]: I believe this is an argument encoded issue, as this example program I think demonstrates:

```
$ echo 'def main (as : List String) : IO Unit := IO.println (repr as)' | lean --stdin --run "«path with spaces»"
[]

Comment (Kha) [reply tests/pkg/spaces/test.sh]: I think that's a different `--run` bug, at least on Linux. Could you try
```
echo 'def main (as : List String) : IO Unit := IO.println (repr as)' | lean --stdin --run '«path with spaces»' '«path with spaces»'
```
?

Comment (Kha) [reply tests/pkg/spaces/test.sh]: #4601

**Comment by Kha** [comment]
Well, now the test isn't broken even without escaping. It might be some weird edge case like "`CreateProcess` with `A\B C.exe "an arg"` works, but only if `A\B` does not exist by itself" :shrug: .

**Comment by Kha** [comment]
Hah, successfully failed the test. Now to add back the fix.

> error: unspecified system_category error (error code: 193)
